File: lib/core/presets.dart
enum Preset { subtle, standard, aggressive }

class StriptParams {
  final double noiseFraction;
  final int noiseStrength;
  final double resizeScale;
  final int jpegQuality;

  const StriptParams({
    this.noiseFraction = 0.03,
    this.noiseStrength = 1,
    this.resizeScale = 0.998,
    this.jpegQuality = 97,
  });

  static const presetParams = {
    Preset.subtle: StriptParams(
      noiseFraction: 0.01,
      noiseStrength: 1,
      resizeScale: 0.999,
      jpegQuality: 99,
    ),
    Preset.standard: StriptParams(
      noiseFraction: 0.03,
      noiseStrength: 1,
      resizeScale: 0.998,
      jpegQuality: 97,
    ),
    Preset.aggressive: StriptParams(
      noiseFraction: 0.06,
      noiseStrength: 2,
      resizeScale: 0.995,
      jpegQuality: 92,
    ),
  };

  StriptParams copyWith({
    double? noiseFraction,
    int? noiseStrength,
    double? resizeScale,
    int? jpegQuality,
  }) {
    return StriptParams(
      noiseFraction: noiseFraction ?? this.noiseFraction,
      noiseStrength: noiseStrength ?? this.noiseStrength,
      resizeScale: resizeScale ?? this.resizeScale,
      jpegQuality: jpegQuality ?? this.jpegQuality,
    );
  }

  @override
  bool operator ==(Object other) =>
      other is StriptParams &&
      noiseFraction == other.noiseFraction &&
      noiseStrength == other.noiseStrength &&
      resizeScale == other.resizeScale &&
      jpegQuality == other.jpegQuality;

  @override
  int get hashCode => Object.hash(noiseFraction, noiseStrength, resizeScale, jpegQuality);
}

File: src-tauri/src/engine/presets.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
pub enum Preset {
    Subtle,
    Standard,
    Aggressive,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ProcessParams {
    pub noise_fraction: f64,
    pub noise_strength: u8,
    pub resize_scale: f64,
    pub jpeg_quality: u8,
}

impl Default for ProcessParams {
    fn default() -> Self {
        Self::standard()
    }
}

impl ProcessParams {
    pub fn subtle() -> Self {
        Self {
            noise_fraction: 0.01,
            noise_strength: 1,
            resize_scale: 0.999,
            jpeg_quality: 99,
        }
    }

    pub fn standard() -> Self {
        Self {
            noise_fraction: 0.03,
            noise_strength: 1,
            resize_scale: 0.998,
            jpeg_quality: 97,
        }
    }

    pub fn aggressive() -> Self {
        Self {
            noise_fraction: 0.06,
            noise_strength: 2,
            resize_scale: 0.995,
            jpeg_quality: 92,
        }
    }

    pub fn from_preset(preset: Preset) -> Self {
        match preset {
            Preset::Subtle => Self::subtle(),
            Preset::Standard => Self::standard(),
            Preset::Aggressive => Self::aggressive(),
        }
    }
}
